### PR TITLE
blastco armory is now completely burglarproof (TM)

### PR DESCRIPTION
**Pull request by Spacedong**
Blastco anti-burglar payloads will now instantly detonate upon sensing a mob in the armory. They will despawn once blastco is bought legitimately. Fixes issue #3254. Please codecheck this before merging.

### CODE REVIEW

**Comment by tortellinitony** [comment]
why not just make the blast doors invincible?

**Comment by Spacedong** [comment]
but then its still not completely burglar proof

File: code/game/machinery/syndicatebomb.dm [hunk 1]
```diff
@@ -353,4 +354,8 @@
 		message_admins(adminlog)
 		log_game(adminlog)
 	explosion(get_turf(src),20,20,20, flame_range = 20)
-	qdel(src)
\ No newline at end of file
+	qdel(src)
+
+/obj/machinery/blastco_antiburglar/HasProximity(atom/movable/AM as mob|obj)
```
Comment (ghost) [inline code/game/machinery/syndicatebomb.dm:359]: Won't this explode right on the start?

**Comment by Spacedong** [comment]
why would it? it only detects if you're within a tile from it, like a portable flasher or a proximity grenade

File: code/game/machinery/syndicatebomb.dm (hunk 1 above)
Comment (Spacedong) [inline code/game/machinery/syndicatebomb.dm]: why would it? it only detects if you're within a tile from it, like a portable flasher or a proximity grenade

**Comment by Spacedong** [comment]
@ArcLumin add some tags on this bitch

**Comment by ArcLumin** [comment]
good enough for you, dong?

File: code/game/machinery/syndicatebomb.dm (hunk 1 above)
Comment (ghost) [reply code/game/machinery/syndicatebomb.dm:359]: Can a ghost explode it?

Comment (Spacedong) [inline code/game/machinery/syndicatebomb.dm]: of course not

Comment (ghost) [reply code/game/machinery/syndicatebomb.dm:359]: ARE YOU SURE ABOUT THAT?

Comment (Spacedong) [inline code/game/machinery/syndicatebomb.dm]: yes https://vgy.me/nb1tTt.png

**Comment by MethMedic** [comment]
Can't wait for ghosts to blow these up cuz dongcode

**Comment by MethMedic** [comment]
Also why don't we have a dongcode tag

**Comment by Spacedong** [comment]
I just took a pic of me as a ghost standing next to the payload and not blowing up you reatard

**Comment by MethMedic** [comment]
You underestimate dongcode

**Comment by KittyKathy** [comment]
You know it would have made more sense when they originally coded it to make the room empty and when they buy blastco it would teleport(spawn) the stuff in the room. Can't steal what isn't there right?

**Comment by GuyonBroadway** [comment]
I dunno, I know Hippe doesn't really have "lore" but I like the idea that Blastco are a bunch of crazy fucking bastards who will find a way to cram explosives into everything.

**Comment by SpankMasterson** [comment]
sure +1

**Comment by McBawbaggings** [comment]
+1

**Comment by ArcLumin** [comment]
Merging since head approval

**Comment by ArcLumin** [comment]
DONG YOU SAID IN THE CODE REVIEW YOU TESTED THIS AAAAAAAA

**Comment by Spacedong** [comment]
local man merges shit that wasnt even codechecked

**Comment by ArcLumin** [comment]
You said it was codechecked by McBaw

Also that talk up there was your codecheck, which you said was gucci

**Comment by Pyko1** [comment]
Seriously @spaceddong, compile and test your code yourself before making a PR.

**Comment by Pyko1** [comment]
also wanna know whats the issue?

`
/datum/uplink_item/equipaccess/blastco/buy(obj/item/device/uplink/U, mob/user)
	..()
	for(var/obj/machinery/blastco_antibuurglar in world)
		qdel()
`
`/obj/machinery/blastco_antiburglar/ex_act(severity, target) //Little boom can chain a big boom`

for(var/obj/machinery/blastco_antibu**u**rglar in world)

**Comment by Pyko1** [comment]
Emphasis on the double **U** on dongcode.

**Comment by ArcLumin** [comment]
For someone who looks through the code for typos on a daily basis he doesn't check his own much

**Pull request merged**